crates/polars-ops/src/chunked_array/strings/namespace.rs

```diff
@@ -216,6 +216,35 @@ pub trait StringNameSpaceImpl: AsString {
         }
     }
 
+    /// Check if strings starts with a substring
+    fn starts_with(&self, sub: &str) -> BooleanChunked {
+        let ca = self.as_string();
+
+        let iter = ca.downcast_iter().map(|arr| {
+            let out: <BooleanType as PolarsDataType>::Array = arr
```

Comment:
If data buffers are empty, we know that all views are inlined and we don't have to branch in the `view.starts_with`, I think we should have a fast path for that.

Reply:
Wouldn't the branch predictor make this basically free as it will be going into the same branch every time?

Reply:
I could block auto-vectorization, as it still has to do the check. But that's something that we could benchmark.